numpy/core/src/multiarray/buffer.c

```diff
@@ -483,7 +483,25 @@ _buffer_info_new(PyObject *obj)
         goto fail;
     }
 
-    if (PyArray_IsScalar(obj, Generic)) {
+    if (PyArray_IsScalar(obj, Datetime) || PyArray_IsScalar(obj, Timedelta)) {
+        info->ndim = 1;
+        /*
+         * Special case datetime64 scalars to remain backward compatible.
+         * This will change in a future version.
+         * Note arrays of datetime64 and strutured arrays with datetime64
+         * fields will not hit this code path and are currently unsupported
+         * in _buffer_format_string.
+         */
+        _append_char(&fmt, 'B');
+        _append_char(&fmt, '\0');
+        info->shape = malloc(sizeof(Py_ssize_t) * 2);
```

Comment:
malloc can fail - this needs to call `PyErr_NoMemory()` if so

Reply:
fixed